nix/egil-scim-client.nix

```diff
@@ -0,0 +1,100 @@
+{ version
+, homepage
+, downloadPage
+, changelog
+, maintainers
+, platforms
+, isDebugBuild ? false
+, doCheck ? true
+}:
+{ lib
+, boost
+, cmake
+, curl
+, filterSource
+, openldap
+, stdenv
+, ...
+}:
+
+let
+  inherit (lib)
+    optionalString
+    optionals
+  ;
+
+  packageName = "egil-scim-client";
+  exeName = "EgilSCIMClient";
+
+  suffix = optionalString isDebugBuild "-debug";
+  pname = "${packageName}${suffix}";
+  exePath = "/bin/${exeName}${suffix}";
+in
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = filterSource {
+    src = ./..;
+    filesToKeep = [ "CMakeLists.txt" ];
+    directoriesToKeep = [ "/src" ];
+    name = packageName;
+  };
```

Comment:
Last time we tried using this function it did not work. You sure that it evaulates?

Reply:
Yes, I am aware of that problem but I am sure that this works. Maybe it's because this is not `builtins.filterSource` but rather a function that I [wrote][^1] that uses `nixpkgs.lib.cleanSourceWith`.

[^1]: https://git.sr.ht/~ilkecan/source-utils/tree/master/item/default.nix#L13

Reply:
Can you please upstream it to nixpkgs? I think it would be neat

Reply:
I think [nix-filter](https://github.com/numtide/nix-filter) is a more mature library in line with what I had in mind when I wrote that. I wanted a whitelist approach as opposed to ones like using `.gitignore`. So I will try to replace the custom code with "nix-filter" if I don't encounter any problem.

Reply:
Resolved with becba23f88230dc5257b2323d3bb724aeb60d8f5